### PR TITLE
docs(ingest): add a guide for writing sources

### PR DESCRIPTION
**Pull request by hsheth2**
Also upgrade docusaurus and fix some site generation bugs.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by shirshanka** [comment]
@hsheth2 : looks like you need to rebase this

File: metadata-ingestion/adding-source.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+### 4. Set up the dependencies
+
+Declare the source's pip dependencies in the `plugins` variable of the [setup script](./setup.py).
+
+### 5. Enable discoverability
+
+Declare the source under the `entry_points` variable of the [setup script](./setup.py). This enables the source to be listed when running `datahub check plugins`, and sets up the source's shortened alias for use in recipes.
+
+### 6. Write some tests
```
Comment (jjoyce0510) [inline metadata-ingestion/adding-source.md:31]: minor nit: Write tests

File: metadata-ingestion/adding-source.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Tests go in the `tests` directory. We use the [pytest framework](https://pytest.org/).
+
+### 7. Write some docs
```
Comment (jjoyce0510) [inline metadata-ingestion/adding-source.md:35]: Nit: Write docs

File: metadata-ingestion/developing.md [hunk 3]
```diff
@@ -83,6 +83,8 @@ The syntax for installing plugins is slightly different in development. For exam
 
 Contributions welcome!
 
+Also take a look at the guide to [adding a source](./adding-source.md).
```
Comment (jjoyce0510) [inline metadata-ingestion/developing.md:86]: Awesome. Thank you!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**